File: 1448.rb
# Definition for a binary tree node.
# class TreeNode
#     attr_accessor :val, :left, :right
#     def initialize(val = 0, left = nil, right = nil)
#         @val = val
#         @left = left
#         @right = right
#     end
# end
# @param {TreeNode} node
# @return {Integer}
def good_nodes(node, max = -10001)
  count = 0
  if node.val >= max
    count = 1
    max = node.val
  end
  count += good_nodes(node.left, max) if node.left
  count += good_nodes(node.right, max) if node.right
  count
end

File: 1448.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} node
 * @return {number}
 */
function goodNodes(node, max = -10000) {
  let count = 0;
  if (node.val >= max) {
    count++;
    max = node.val;
  }
  if (node.left)
    count += goodNodes(node.left, max);
  if (node.right)
    count += goodNodes(node.right, max);
  return count;
}
